ixmp4/data/api/base.py

```diff
@@ -203,34 +215,38 @@ def _handle_response(
             raise self.get_remote_exception(res, res.status_code)
         else:
             try:
-                return res.json()
+                json_decoded: dict | list = res.json()
+                return json_decoded
             except JSONDecodeError:
                 if res.status_code < 300 and res.text == "":
                     return None
             except ValueError:
                 pass
             raise UnknownApiError(res.text)
 
-    def _get_by_id(self, id: int, *args, **kwargs) -> dict[str, Any]:
+    def _get_by_id(self, id: int) -> dict[str, Any]:
         # we can assume this type on create endpoints
-        return self._request("GET", self.prefix + str(id) + "/", **kwargs)  # type: ignore
+        return self._request("GET", self.prefix + str(id) + "/")  # type: ignore[return-value]
 
     def _request_enumeration(
         self,
         table: bool = False,
         params: dict | None = None,
         json: dict | None = None,
-    ):
+    ) -> dict | list:
         """Convenience method for requests to the enumeration endpoint."""
         if params is None:
             params = {}
 
+        # TODO instead of assuming this, wouldn't it be better to never return None and
+        # handle this as an exception or some such in _request()?
+        # we can assume these types on enumeration endpoints
```

Comment:
Hm, we return `None` if the response was successful but empty:
```py
            try:
                return res.json()
            except JSONDecodeError:
                if res.status_code < 300 and res.text == "":
                    return None
```

I feel like the repositories should be able to handle empty success responses, but im not sure if this is actually used somewhere right now...

Reply:
Okay, I'm fine with keeping the `type: ignore` for now. I'll link this conversation for future reference, though :)